src/backend/aspen/app/views/phylo_trees.py

```diff
@@ -105,6 +106,10 @@ def phylo_trees():
                 "phylo_tree_id": None,
                 "name": phylo_run.name or generate_tree_name_from_template(phylo_run),
             }
+            if phylo_run.start_datetime and phylo_run.start_datetime < (
```

Comment:
The one thing I'm unclear on is if we get to this part of the code, we are guaranteed that the phylo tree is not "completed" right? I think that's due to the `if isinstance(output, PhyloTree):` section above if I'm reading it correctly? We check each thing for a PhyloTree output, and if it is one, then it's implicitly completed because we have the tree -- am I correct?

If so, then this all looks good. If I'm misunderstanding the above, then my one worry is something that should somehow be complete still getting marked as FAILED just because it ran super slow.

Reply:
Yep, this is exactly what's happening.